tensorflow/compiler/xla/service/gpu/gpu_fusible.cc

```diff
@@ -131,5 +131,16 @@ bool ShapesCompatibleForMultiOutputFusion(const HloInstruction& instr1,
                                              get_loop_shape(instr_2));
 }
 
+bool IsMultiOutputFusible(const HloInstruction& instr) {
+  // We can fuse reduces and loop fusions. Elementwise instructions can be fused
+  // with any other instruction.
+  // TODO(b/112957171): This should use the same isFusible logic as
```

Comment:
Shall I instead replace the entire check with "return IsInputFusible(instr) || IsLoopFusible(instr);"?

Reply:
> return IsInputFusible(instr) || IsLoopFusible(instr);
We need to use IsInputFusibleReduction. Scatter cannot be the root of a multi-output fusion because its emitter doesn't support it. Please add this as a code comment.
Note that DoProducerConsumerMultiOutputFusion has to do the same check, i.e. can this op be the root of a multi-output fusion. Hence, it makes sense to extract this function and use it in both places.

Reply:
Would it help to have a function that returns whether the emitter supports a certain combination of operations? Then, as support is added, this function can be updated instead of modifying all the various checks throughout the code base.

Reply:
> Would it help to have a function that returns whether the emitter supports a certain combination of operations?

IsInputFusibleReduction and IsInputFusibleScatter serve exactly this purpose. If we would add support for sibling multi-output fusion to the Scatter emitter we would only need to touch the latter.